ping-thing-client.mjs

```diff
@@ -139,11 +128,14 @@ async function pingThing() {
 
         while (true) {
           try {
-            await mSendTransaction(transactionSignedWithFeePayer, {
-              commitment: "confirmed",
-              maxRetries: 0n,
-              skipPreflight: true,
-            });
+            [slotSent] = await Promise.all([
+              getNextSlot(),
+              mSendTransaction(transactionSignedWithFeePayer, {
+                commitment: "confirmed",
+                maxRetries: 0n,
+                skipPreflight: true,
+              }),
+            ]);
```

Comment:
Fetch the sent slot as near to the actual send as possible. I don't know if you want the slot on the _leading_ edge (ie. you don't mind the actual send being one slot later) or the trailing edge (ie. you don't mind it being earlier). I fetch them in parallel here, and the `getNextSlot()` function really does wait for the _next_ slot (it's not a read-through cache like the blockhash iterator).

Reply:
Know what, I updated this PR because I think I see what's going on here. You want the sentSlot to live outside this loop and be computed:

1. Once.
2. Before even trying to send.